kura/org.eclipse.kura.ble.provider/src/main/java/org/eclipse/kura/internal/ble/BluetoothLeGattCharacteristicImpl.java

```diff
@@ -72,44 +75,65 @@ public BluetoothLeGattDescriptor findDescriptor(UUID uuid, long timeout)
     public byte[] readValue() throws KuraBluetoothIOException {
         byte[] value;
         try {
-            value = BluetoothLeGattCharacteristicImpl.this.characteristic.readValue();
-        } catch (BluetoothException e) {
+            value = this.characteristic.readValue(null);
+        } catch (DBusException e) {
             throw new KuraBluetoothIOException(e, "Read characteristic value failed");
         }
         return value;
     }
 
     @Override
     public void enableValueNotifications(Consumer<byte[]> callback) throws KuraBluetoothNotificationException {
-        BluetoothLeNotification<byte[]> notification = new BluetoothLeNotification<>(callback);
         try {
-            this.characteristic.enableValueNotifications(notification);
-        } catch (Exception e) {
+            DeviceManager.getInstance().registerPropertyHandler(new AbstractPropertiesChangedHandler() {
+                @Override
+                public void handle(PropertiesChanged props) {
+                    if (props != null) {
+                        if (!props.getPath().contains(BluetoothLeGattCharacteristicImpl.this.characteristic.getDbusPath())) {
+                            return;
+                        }
+
+                        for (Map.Entry<String, Variant<?>> entry : props.getPropertiesChanged().entrySet()) {
+                            if (entry.getKey().equals("Value")) {
+                                Object obj = entry.getValue().getValue();
+                                byte[] val = (byte[]) obj;
+                                callback.accept(val);
+                            }
+                        }
+                    }
+                }
+            });
+            this.characteristic.startNotify();
+        } catch (DBusException e) {
```

Comment:
Why we need to call `startNotify`?

Reply:
Once we setup the PropertyHandler you have to set the characteristic notifying on change in order to receive anything on the callback. The previous TinyB enableValueNotifications() method did the same internally.